### PR TITLE
Productivity suite demo

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: productivity-suite/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Productivity Suite Demo
+
+![app](./app-screenshot.png)
+
+This project demonstrates what we've called the "Migration Piercing Strategy", which consists in a strategy aimed to the incremental migration of a legacy application to a server side rendering model implemented at the edge via Cloudflare Workers.
+
+To read more about it see the [_TITLE TBD_ blog post](https://blog.cloudflare.com/better-micro-frontends).
```
Comment (dario-piotrowicz) [inline productivity-suite/README.md:7]: I guess we'll keep this as is for now?

File: productivity-suite/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  ```
+
+- Serve the fragments:
+
+  ```
+  cd app/fragments/fragmentName
+  npm run dev
+  ```
+
+  where `fragmentName` is `login`, `todo-lists` and `todos`
```
Comment (dario-piotrowicz) [inline productivity-suite/README.md:84]:
```suggestion
  where `fragmentName` is `login`, `todo-lists`, `todos` and `news`
```

File: productivity-suite/app/fragments/login/src/components/LoginForm.css [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+@keyframes spin-animation {
+    from {
+        transform: rotate(0deg);
+    }
+    to {
+        transform: rotate(360deg);
+    }
+}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/login/src/components/LoginForm.css:142]:
```suggestion
}

```

File: productivity-suite/app/fragments/news/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# SolidStart Hackernews Example
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/news/README.md:1]: do we want to keep this README?

File: productivity-suite/app/fragments/todo-lists/src/components/ListsCarouselAnimations.css [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            transform: scale(var(--tertiary-list-card-scale)) translateX(3rem);
+            opacity: 0;
+        }
+    }
+    @keyframes move-next-next-list-right {
+        from, to {
+            opacity: 0;
+        }
+    }
+}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todo-lists/src/components/ListsCarouselAnimations.css:143]:
```suggestion
}

```

File: productivity-suite/app/fragments/todo-lists/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+		"isolatedModules": true,
+		"noEmit": true,
+		"jsxImportSource": "@builder.io/qwik",
+		"jsx": "react-jsx",
+		"typeRoots": ["./src/types"],
+		// "paths": {
+		// 	"~shared": [
+		// 	  "../../_shared/*"
+		// 	]
+		// }
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todo-lists/tsconfig.json:24]: do we want to keep this in?

File: productivity-suite/app/fragments/todos/src/components/Header.css [hunk 7]
```diff
@@ -0,0 +1,10 @@
+.todo-mvc .new-todo {
+    padding: 1rem 1rem 1rem 3.5rem;
+	border: none;
+	background: rgba(0, 0, 0, 0.003);
+	box-shadow: inset 0 -2px 1px rgba(0,0,0,0.03);
+}
+
+.todo-mvc .new-todo:focus {
+	outline: 1px solid #00000075;
+}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todos/src/components/Header.css:10]:
```suggestion
}

```

File: productivity-suite/app/fragments/todos/src/components/ListOfTodos.css [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+.todo-mvc-todos-list li.editing:last-child {
+	margin-bottom: -1px;
+}
+
+.hidden-label {
+	font-size: 0px;
+	position: absolute;
+	top: 0.5rem;
+	left: 0.9rem;
+}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todos/src/components/ListOfTodos.css:138]:
```suggestion
}

```

File: productivity-suite/app/fragments/todos/src/components/ToggleAllTodosButton.css [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	height: 1px;
+	border: none;
+	opacity: 0;
+	position: absolute;
+	right: 100%;
+	bottom: 100%;
+}
+
+.todo-mvc-toggle-all + label {
+	font-size: 0px;
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todos/src/components/ToggleAllTodosButton.css:13]:
```suggestion
    font-size: 0px;
```

File: productivity-suite/app/fragments/todos/src/components/ToggleAllTodosButton.css [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    position: absolute;
+    top: 0.9rem;
+    left: 0.2rem;
+}
+
+.todo-mvc-toggle-all + label:before {
+    content: ">";
```
Comment (dario-piotrowicz) [inline productivity-suite/app/fragments/todos/src/components/ToggleAllTodosButton.css:20]:
```suggestion
	content: ">";
```

File: productivity-suite/app/legacy-app/src/components/layout.css [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+  top: 100%;
+  width: 100%;
+  z-index: 10;
+  display: flex;
+  flex-direction: column;
+  animation: slide-in .1s ease-in-out forwards;
+}
+
+@keyframes slide-in {
+	0% {
```
Comment (dario-piotrowicz) [inline productivity-suite/app/legacy-app/src/components/layout.css:76]:
```suggestion
  0% {
```

File: productivity-suite/app/legacy-app/src/components/layout.css [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+    transform: translateY(-1rem);
+    opacity: 0;
+	}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/legacy-app/src/components/layout.css:79]:
```suggestion
  }
```

File: productivity-suite/app/legacy-app/src/components/layout.css [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+	100% {
```
Comment (dario-piotrowicz) [inline productivity-suite/app/legacy-app/src/components/layout.css:80]:
```suggestion
  100% {
```

File: productivity-suite/app/legacy-app/src/components/layout.css [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+    transform: translateY(0);
+    opacity: 1;
+	}
```
Comment (dario-piotrowicz) [inline productivity-suite/app/legacy-app/src/components/layout.css:83]:
```suggestion
  }
```

File: productivity-suite/app/shared/src/api.ts [hunk 15]
```diff
@@ -0,0 +1,225 @@
+// Note:
+//  The following file, for the purposed of this demo just interact with cookies instead
```
Comment (dario-piotrowicz) [inline productivity-suite/app/shared/src/api.ts:2]:
```suggestion
//  The following file, for the purposes of this demo just interact with cookies instead
```

**Review by dario-piotrowicz — APPROVED** [review]

File: productivity-suite/scripts/start-dev.js [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+    });
+}
+
+function sleep(delay) {
+    return new Promise(resolve => setTimeout(resolve, delay));
+}
+
+function log(message) {
+   console.log(`\x1b[34m ${message}\n \x1b[0m`);
+}
```
Comment (dario-piotrowicz) [inline productivity-suite/scripts/start-dev.js:86]:
```suggestion
}

```

**Comment by petebacondarwin** [comment]
Merged as df50b60cfff7bc299cf70ecfe8f7826ec9313b84

**Pull request closed without merging**